### ISSUE TITLE
I connected other players in my app by referring to video_player, but when he was on the andriod 12 mobile phone, a black screen occurred when the player switched to full screen

### ISSUE THREAD

**Issue by SSunWuKong** [flutter/flutter#95724]
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. ... 
    on the andriod 12 mobile phone  a black screen occurred when the player switched to full screen。 

There will be a black screen on the player, but there is still sound 

only on andriod 12 have this problem

```
E/BufferQueueProducer(16667): [SurfaceTexture-0-16667-145] cancelBuffer: BufferQueue has been abandoned
I/chatty  (16667): uid=10234(com.example.cook_it) JNISurfaceTextu identical 10 lines
E/BufferQueueProducer(16667): [SurfaceTexture-0-16667-145] cancelBuffer: BufferQueue has been abandoned
D/SurfaceUtils(16667): disconnecting from surface 0x797bec7010, reason disconnectFromSurface
I/ExoPlayerImpl(16667): Init 8b84e21 [ExoPlayerLib/2.14.1] [HWCLT, CLT-L29, HUAWEI, 28]
V/AudioManager(16667): getStreamVolume  treamType: 3
V/AudioManager(16667): isStreamMute   streamType: 3
V/AudioManager(16667): getStreamMaxVolume  treamType: 3
E/        (16667): [ZeroHung]zrhung_get_config: Get config failed for wp[0x0008]
I/OMXClient(16667): IOmx service obtained
I/ACodec  (16667): In onAllocateComponent create compenent, codec name: OMX.hisi.video.decoder.avc
D/SurfaceUtils(16667): connecting to surface 0x797c027010, reason connectToSurface
I/MediaCodec(16667): [OMX.hisi.video.decoder.avc] setting surface generation to 17067154
D/SurfaceUtils(16667): disconnecting from surface 0x797c027010, reason connectToSurface(reconnect)
D/SurfaceUtils(16667): connecting to surface 0x797c027010, reason connectToSurface(reconnect)
E/ACodec  (16667): [OMX.hisi.video.decoder.avc] setPortMode on output to DynamicANWBuffer failed w/ err -2147483648
I/HwExtendedCodec(16667): mime is [video/avc] at setVideoFormat
I/ACodec  (16667): codec does not support config priority (err -22)
```
2

**Comment by darshankawar** [comment]
@SSunWuKong 
Can you provide `flutter doctor -v` , plugin version and [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example
) along with a video (if possible) that shows this behavior ?

**Comment by SSunWuKong** [comment]
> @SSunWuKong Can you provide `flutter doctor -v` , plugin version and [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) along with a video (if possible) that shows this behavior ?

   • Flutter version 2.5.0 
I also upgraded to the latest version to try. I just imitated (video_player) and connected a player with platformView. When I cut the player to full screen, I changed the size of platformView to full screen, and then the above error occurred. , The video playback screen turned into a black screen, there is no picture, but there is sound, even if I change the size of platformView back, the problem still exists, but this problem only occurs on the mobile phone with andriod 12 system installed,

 some important logs like：
E/BufferQueueProducer(16667): [SurfaceTexture-0-16667-145] cancelBuffer: BufferQueue has been abandoned
D/SurfaceUtils(16667): disconnecting from surface 0x797bec7010, reason disconnectFromSurface


I can’t find the reason, why when I change the size of the player, the SurfaceView. will be released,


Simple demo. I will try to make，
